File: ObjectiveC/FlexChart101/FlexChart101/ThemingController.m
//
//  ThemingController.m
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "ThemingController.h"
@import XuniFlexChartDynamicKit;
#import "ChartData.h"

@interface ThemingController (){
    NSMutableArray *pickerData;
}
@property (weak, nonatomic) IBOutlet FlexChart *chart;
@property (weak, nonatomic) IBOutlet UIPickerView *picker;
@end

@implementation ThemingController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self setTitle:NSLocalizedString(@"Theming", nil)];
    
    pickerData =[[NSMutableArray alloc] initWithObjects:@"Standard", @"Cocoa", @"Coral", @"Dark", @"HighContrast", @"Light", @"Midnight", @"Minimal", @"Modern", @"Organic", @"Slate", @"Zen", @"Cyborg", @"Superhero", @"Flatly", @"Darkly", @"Cerulean", nil];

    self.picker.delegate = self;
    
    FlexChart *chart = self.chart;
    XuniSeries *sales = [[XuniSeries alloc] initForChart:chart binding:@"sales, sales" name:@"Sales"];
    XuniSeries *expenses = [[XuniSeries alloc] initForChart:chart binding:@"expenses, expenses" name:@"Expenses"];
    XuniSeries *downloads = [[XuniSeries alloc] initForChart:chart binding:@"downloads, downloads" name:@"Downloads"];
    
    [chart.series addObject:sales];
    [chart.series addObject:expenses];
    [chart.series addObject:downloads];
    
    chart.bindingX = @"name";
    chart.itemsSource = [ChartData demoData];
    
}


- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView{
    return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    return [pickerData count];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    FlexChart *chart = self.chart;
     switch (row) {
        case 0:
            chart.palette = [XuniPalettes standard];
            break;
        case 1:
            chart.palette = [XuniPalettes cocoa];
            break;
        case 2:
            chart.palette = [XuniPalettes coral];
            break;
        case 3:
            chart.palette = [XuniPalettes dark];
            break;
        case 4:
            chart.palette = [XuniPalettes highcontrast];
            break;
        case 5:
            chart.palette = [XuniPalettes light];
            break;
        case 6:
            chart.palette = [XuniPalettes midnight];
            break;
        case 7:
            chart.palette = [XuniPalettes minimal];
            break;
        case 8:
            chart.palette = [XuniPalettes modern];
            break;
        case 9:
            chart.palette = [XuniPalettes organic];
            break;
        case 10:
            chart.palette = [XuniPalettes slate];
            break;
        case 11:
            chart.palette = [XuniPalettes zen];
            break;
        case 12:
            chart.palette = [XuniPalettes cyborg];
            break;
        case 13:
            chart.palette = [XuniPalettes superhero];
            break;
        case 14:
            chart.palette = [XuniPalettes flatly];
            break;
        case 15:
            chart.palette = [XuniPalettes darkly];
            break;
        case 16:
            chart.palette = [XuniPalettes cerulean];
            break;
        default:
            break;
    }
}

- (NSString*)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return [pickerData objectAtIndex:row];
}

@end

File: Swift/FlexChart101/FlexChart101/ThemingController.swift
//
//  ThemingController.swift
//  FlexChart101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit
import FlexChartKit

class ThemingController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate {
    
    var _chart = FlexChart()
    var _pickerView = UIPickerView()
    var _pickerData = NSMutableArray()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Do any additional setup after loading the view.
        
        _pickerData = ["Standard", "Cocoa", "Coral", "Dark", "HighContrast", "Light", "Midnight", "Minimal", "Modern", "Organic", "Slate", "Zen", "Cyborg", "Superhero", "Flatly", "Darkly", "Cerulean"]
        _pickerView.delegate = self
        _pickerView.showsSelectionIndicator = true
        _pickerView.hidden = false
        
        _chart.bindingX = "name"
        
        let sales = XuniSeries(forChart: _chart, binding: "sales, sales", name: "Sales")
        let expenses = XuniSeries(forChart: _chart, binding: "expenses, expenses", name: "Expenses")
        let downloads = XuniSeries(forChart: _chart, binding: "downloads, downloads", name: "Downloads")
        
        _chart.series.addObject(sales)
        _chart.series.addObject(expenses)
        _chart.series.addObject(downloads)
        
        _chart.itemsSource = ChartData.demoData()
        
        _chart.legend.orientation = XuniChartLegendOrientation.Auto
        _chart.legend.position = XuniChartLegendPosition.Auto
        _chart.tooltip.isVisible = true
        _chart.axisX.labelsVisible = true
        _chart.axisY.labelsVisible = true
        
        self.view.addSubview(_chart)
        self.view.addSubview(_pickerView)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        _chart.frame = CGRectMake(0, 206, self.view.bounds.size.width, (self.view.bounds.size.height-206))
        _pickerView.frame = CGRectMake(self.view.bounds.size.width/4, 44, self.view.bounds.size.width/2, 162)
    }
    func numberOfComponentsInPickerView(pickerView: UIPickerView) -> Int {
        return 1
    }
    func pickerView(pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return _pickerData.count
    }
    func pickerView(pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        switch (row) {
        case 0:
            _chart.palette = XuniPalettes.standard()
        case 1:
            _chart.palette = XuniPalettes.cocoa()
        case 2:
            _chart.palette = XuniPalettes.coral()
        case 3:
            _chart.palette = XuniPalettes.dark()
        case 4:
            _chart.palette = XuniPalettes.highcontrast()
        case 5:
            _chart.palette = XuniPalettes.light()
        case 6:
            _chart.palette = XuniPalettes.midnight()
        case 7:
            _chart.palette = XuniPalettes.minimal()
        case 8:
            _chart.palette = XuniPalettes.modern()
        case 9:
            _chart.palette = XuniPalettes.organic()
        case 10:
            _chart.palette = XuniPalettes.slate()
        case 11:
            _chart.palette = XuniPalettes.zen()
        case 12:
            _chart.palette = XuniPalettes.cyborg()
        case 13:
            _chart.palette = XuniPalettes.superhero()
        case 14:
            _chart.palette = XuniPalettes.flatly()
        case 15:
            _chart.palette = XuniPalettes.darkly()
        case 16:
            _chart.palette = XuniPalettes.cerulean()
        default:
            break
        }
    }
    func pickerView(pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String! {
        return _pickerData.objectAtIndex(row) as! String
    }
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
